src/lib/support/tests/TestCHIPArgParser.cpp

```diff
@@ -188,7 +188,7 @@ static void SimpleParseTest_SingleLongOption()
     ClearCallbackRecords();
     PrintArgError = HandleArgError;
 
-    res = ParseArgs(__FUNCTION__, argc, (char **) argv, optionSets, HandleNonOptionArgs);
+    res = ParseArgs(__FUNCTION__, argc, const_cast<char **>(argv), optionSets, HandleNonOptionArgs);
```

Comment:
It seems to me that the right fix here is to remove the `const` from the declaration of `argv` in this test.  Since the type of `argv` in general is `char**` or so, it's valid for arg parsers to actually change the strings.  Passing static const memory with `const` cast away to something like that is flirting with undefined behavior (by assuming that `ParseArgs` in fact does _not_ do any such string modification) more than I'm comfortable with.

Reply:
These are string constants, we'd have to copy them to do this.

Reply:
Or declare them as arrays.

Reply:
Hrm, right. Of course option B is to change ParseArgs to take `const char**`.  Maybe a separate PR to do that, on top of this one, if it's viable?

Reply:
Just tried that, and then we have the problem that getopt_long takes a

  `char * const argv[]`

so it just moves the casts there.

Reply:
It seems like the best option might just be to copy the data, unfortunately.

Reply:
Let's not add more to this PR; maybe just file a new issue to sort this part out.

Reply:
Ok, uploaded https://github.com/mspang/connectedhomeip/pull/new/for-chip/const-dupe-args